src/java/org/apache/cassandra/db/commitlog/CommitLog.java

```diff
@@ -266,11 +260,8 @@ public Map<Keyspace, Integer> recoverFiles(ColumnFamilyStore.FlushReason flushRe
         CommitLogReplayer replayer = CommitLogReplayer.construct(this, getLocalHostId());
         replayer.replayFiles(clogs);
 
-        /*we need to reload segments with invalid mutations as it may have been updated by the commit log replayer
-          Now, we can do delete removed segments and add new ones or just reload afresh
-          I decided to simply reload as this list is likely to be small, if it exists at all
-         */
-        segmentsWithInvalidMutations = replayer.commitLogReader.segmentsWithInvalidMutations;
+        // we need to reload clogs as new invalid mutations may have been added or old invalid mutations may have been replayed
```

Comment:
It might be me, but I still do not quite understand why is this comment put here.

Reply:
Ahh no this is my bad - fixed.